hilti/toolchain/src/compiler/jit.cc

```diff
@@ -18,6 +23,37 @@
 
 using namespace hilti;
 
+namespace {
+
+std::string readFile(const hilti::rt::filesystem::path& path) {
+    std::ifstream ifs(path);
+    return {(std::istreambuf_iterator<char>(ifs)), (std::istreambuf_iterator<char>())};
+}
+
+hilti::rt::filesystem::path save(const CxxCode& code, const hilti::rt::filesystem::path& id, std::size_t hash) {
+    const std::size_t cc_hash = code.hash();
+
+    const auto cc = hilti::rt::filesystem::temp_directory_path() /
+                    util::fmt("%s_%" PRIx64 "-%" PRIx64 ".cc", id.stem().c_str(), hash, cc_hash);
+
+    HILTI_DEBUG(logging::debug::Jit, util::fmt("writing temporary code for %s to %s", id, cc.filename().native()));
+    const auto file = std::fopen(cc.c_str(), "w");
```

Comment:
Why use the C-style file API here? For consistency, staying with C++ streams seems nicer.